token-metadata/js/test/update.test.ts

```diff
@@ -1422,3 +1421,109 @@ test('Update: Cannot Update pNFT Config with locked token', async (t) => {
     /Cannot update the rule set of a programmable asset that has a delegate/i,
   );
 });
+
+test('Update: programmable config delegate update collection item', async (t) => {
```

Comment:
This is a good point, I will add this. The only caveat is whether we should fail the transaction if you try to update a field you are not allowed or the update should succeed but don't change anything.

Reply:
To me it seems better to fail, otherwise maybe someone tries to use the wrong delegate and thinks they were successful but there was a silent failure.  Although most people probably should and would choose to inspect the metadata after doing something to see if it really updated.

Reply:
Done – now there is a validation of the update args.